KAFKA-4660 : Improve test coverage KafkaStreams

umesh9794 opened the pull request:
@dguy , @mjsax Please review the PR and let me know your comments.

umesh9794 committed:
Rebase: KAFKA-5171 : TC should not accept empty string transactional id

umesh9794 committed:
Merge branch 'trunk' of https://github.com/apache/kafka into mylocal

umesh9794 committed:
KAFKA-4660 : Improve test coverage KafkaStreams

mjsax commented:
@umesh9794 Thanks for the patch. Can you assign the JIRA to yourself?

Build fails due to checkstyle errors. Btw: we plan to remove `KafkaStreams#toString()` (cf. https://cwiki.apache.org/confluence/display/KAFKA/KIP+130%3A+Expose+states+of+active+tasks+to+KafkaStreams+public+API) -- so I am actually not sure, how extensive this test should be. Right now it pretty basic, but might be good enough. \cc @dguy @enothereska @guozhangwang @bbejeck

bbejeck commented on streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java:
```diff
@@ -289,6 +289,16 @@ public void testCannotCleanupWhileRunning() throws Exception {
+        String streamString = streams.toString();
+        streams.close();
+        Assert.assertNotEquals("streamString should not be empty", "", streamString);
+        Assert.assertNotNull("streamString should not be null", streamString);
```
Maybe add an assert checking `streamsString` contains the app-id.   But considering this method will be removed, I'm also fine with it as is.

umesh9794 replied:
Thanks for the review. I added to check that appId is not null or non-empty in `streamsString `

bbejeck commented:
Added one minor comment.

umesh9794 committed:
Merge branch 'trunk' of https://github.com/apache/kafka into mylocal

umesh9794 committed:
KAFKA-4660 : Improve test coverage KafkaStreams

umesh9794 committed:
KAFKA-4660 : Improve test coverage KafkaStreams

umesh9794 commented:
@mjsax Thanks for the review and thanks for the KIP's URL. Fixed the checkstyle error and assigned the JIRA to myself. 

@bbejeck Thanks for the review comments. Added the asserts to check that `streamString` has non-empty and non-null app-Id.

umesh9794 commented:
@mjsax @bbejeck may I request you to review the updates?

bbejeck commented on streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java:
```diff
@@ -289,6 +289,18 @@ public void testCannotCleanupWhileRunning() throws Exception {
+        Assert.assertNotEquals("streamString should not be empty", "", streamString);
+        Assert.assertNotNull("streamString should not be null", streamString);
+        Assert.assertNotEquals("streamString contains non-empty appId", "", appId);
+        Assert.assertNotNull("streamString contains non-null appId", appId);
```
Instead of pulling the value out with a regex, what do you think of `streamsString.contains("appId")`.  Although what you have works as well.

umesh9794 replied:
Current implementation strips the app-id from streamString to check whether it is non-null and non-empty whereas `streamsString.contains("appId")` will just look for the string "appId" which may or may not be the actual app-id. So IMO the current implementation seems more accurate for assert. What do you think?

bbejeck replied:
I think what you have here works fine.  Thanks @umesh9794 for your PR.

bbejeck commented:
@umesh9794 just a minor comment, but otherwise LGTM.

mjsax reviewed (approved):
LGTM.

umesh9794 commented:
@bbejeck @mjsax Thanks for the review.

bbejeck reviewed (commented):
LGTM

umesh9794 commented:
Thanks @bbejeck

guozhangwang commented:
LGTM. Merged to trunk and 0.11.0.
